code/_onclick/item_attack.dm

```diff
@@ -103,7 +103,7 @@
 
 	log_combat(user, M, "attacked", src.name, "(INTENT: [uppertext(user.a_intent)]) (DAMTYPE: [uppertext(damtype)])")
 	add_fingerprint(user)
-
+	return TRUE

```

Comment:
This breaks *a lot* of things. Like a hilarious amount of stuff. Returning TRUE on attack will result in afterattack not getting called which a lot of shit relies on to do its attack effects.
While you could argue that shit relying on afterattack to do it's attack effect is in the wrong here, I would at least like to see the issue somewhat adressed before then.

Reply:
From a quick search, this breaks spraycans, the pride hammer, hot potatoes, plastic knifes and forks, singularity hammer, explosive lances and much more. They will no longer have their special effects when used on mobs.
Incidentally, this also means that all those items don't respect pacifism.

Reply:
The quick and dirty fix would be to migrate the special behaviour of flowers to afterattack, but the proper one would be to move all the above mentioned behaviour to attack instead.

Reply:
attack returning TRUE on a successful attack made perfect sense in my head, it is a shame it wasn't that easy.

How did these items ever work if attack never returned true?

I am thinking quick and dirty right now unless i can find all the locations where a change is needed.